src/integrated/land.jl

```diff
@@ -0,0 +1,669 @@
+                p.soil.turbulent_fluxes.vapor_flux_ice
+            ) +
+            p.snow.snow_cover_fraction * p.snow.turbulent_fluxes.vapor_flux +
+            p.canopy.energy.turbulent_fluxes.transpiration
+
+    end
+    return update_boundary_fluxes!
+end
+
+"""
+    lsm_radiant_energy_fluxes!(p,land::LandModel{FT},
+                                canopy_radiation::Canopy.AbstractRadiationModel{FT},
+                                Y,
+                                t,
+                                ) where {FT}
+
+
+A function which computes the net radiation at the ground surface
+give the canopy radiation model, as well as the outgoing radiation,
+and the net canopy radiation.
+
+Returns the correct radiative fluxes for bare ground in the case
+where the canopy LAI is zero. Note also that this serves the role of
+`canopy_radiant_energy_fluxes!`, which computes the net canopy radiation
+when the Canopy is run in standalone mode.
+"""
+function lsm_radiant_energy_fluxes!(
+    p,
+    land::LandModel{FT},
+    canopy_radiation::Canopy.AbstractRadiationModel{FT},
+    Y,
+    t,
+) where {(FT)}
+    canopy = land.canopy
+    canopy_bc = canopy.boundary_conditions
+    radiation = canopy_bc.radiation
+    earth_param_set = canopy.parameters.earth_param_set
+    _σ = LP.Stefan(earth_param_set)
+    LW_d = p.drivers.LW_d
+    SW_d = p.drivers.SW_d
+    c = LP.light_speed(earth_param_set)
+    h = LP.planck_constant(earth_param_set)
+    N_a = LP.avogadro_constant(earth_param_set)
+    (; λ_γ_PAR, λ_γ_NIR) = canopy_radiation.parameters
+    energy_per_photon_PAR = h * c / λ_γ_PAR
+    energy_per_photon_NIR = h * c / λ_γ_NIR
+    T_canopy =
+        ClimaLand.Canopy.canopy_temperature(canopy.energy, canopy, Y, p, t)
+
+    α_soil_PAR = land.soil.parameters.PAR_albedo
+    α_soil_NIR = land.soil.parameters.NIR_albedo
+    ϵ_soil = land.soil.parameters.emissivity
+    T_soil = ClimaLand.Domains.top_center_to_surface(p.soil.T)
+    α_snow_NIR = land.snow.parameters.α_snow
+    α_snow_PAR = land.snow.parameters.α_snow
+    ϵ_snow = land.snow.parameters.ϵ_snow
+    T_snow = p.snow.T_sfc
+    # in W/m^2
+    LW_d_canopy = p.scratch1
+    LW_u_soil = p.scratch2
+    LW_u_snow = p.scratch3
+    LW_net_canopy = p.canopy.radiative_transfer.LW_n
+    SW_net_canopy = p.canopy.radiative_transfer.SW_n
+    R_net_soil = p.soil.R_n
+    R_net_snow = p.snow.R_n
+    LW_u = p.LW_u
+    SW_u = p.SW_u
+    # in total: d - u = CANOPY_ABS + (1-α_ground)*CANOPY_TRANS
+    # SW_u  = reflected par + reflected nir
+    @. SW_u =
+        energy_per_photon_NIR * N_a * p.canopy.radiative_transfer.nir.refl +
+        energy_per_photon_PAR * N_a * p.canopy.radiative_transfer.par.refl
+
+    # net canopy
+    @. SW_net_canopy =
+        energy_per_photon_NIR * N_a * p.canopy.radiative_transfer.nir.abs +
+        energy_per_photon_PAR * N_a * p.canopy.radiative_transfer.par.abs
+
+
+    # net soil = (1-α)*trans for par and nir
+    @. R_net_soil .=
```

Comment:
It looks like a lot of the radiation update for soil is the same as the update in `src/integrated/soil_canopy_model.jl` - maybe we can pull this out into a shared location

Reply:
I did move one part out into a shared function, and started a file called land_radiation.jl which can contain these shared functions.

I am finding it hard to unify the rest of `lsm_radiant_energy_fluxes!` though, until we fix the sign inconsistency, and I think we should also split it into SW and LW functions. But this feels like a bigger change. I opened this issue: https://github.com/CliMA/ClimaLand.jl/issues/898